modules/javafx.graphics/src/main/java/com/sun/javafx/iio/common/ImageTools.java

```diff
@@ -211,6 +211,21 @@ public static int[] computeDimensions(int sourceWidth, int sourceHeight,
         return new int[]{finalWidth, finalHeight};
     }
 
+    public static void validateMaxDimensions(double width, double height, double scaleFactor) {
+        if (width * scaleFactor > Integer.MAX_VALUE) {
+            throw new IllegalArgumentException("Image width exceeds maximum value");
```

Comment:
You may want to include the values used to reach this conclusion here and with the other 2 exceptions; if these exceptions ever come up, that's the first thing a developer will likely want to know (which may be you if this is posted in an issue).

Reply:
Changed as suggested.